infra/bisector.py

```diff
@@ -0,0 +1,153 @@
+
+  parser.add_argument(
+      '--project_name',
+      help='The name of the project where the bug occured',
+      required=True)
+  parser.add_argument(
+      '--commit_new',
+      help='The newest commit SHA to be bisected',
+      required=True)
+  parser.add_argument(
+      '--commit_old',
+      help='The oldest commit SHA to be bisected',
+      required=True)
+  parser.add_argument(
+      '--fuzz_target', help='the name of the fuzzer to be built', required=True)
+  parser.add_argument(
+      '--testcase', help='the testcase to be reproduced', required=True)
+  parser.add_argument('--engine', default='libfuzzer')
+  parser.add_argument(
+      '--sanitizer',
+      default='address',
+      help='the default is "address"; "dataflow" for "dataflow" engine')
+  parser.add_argument('--architecture', default='x86_64')
+  args = parser.parse_args()
+  build_data = BuildData(args.project_name, args.engine, args.sanitizer,
+                         args.architecture)
+  if os.getcwd() != os.path.dirname(
+      os.path.dirname(os.path.realpath(__file__))):
+    print("Error: bisector.py needs to be run from the OSS-Fuzz home directory")
+    return 1
+  error_sha = bisect(args.commit_old, args.commit_new, args.testcase,
+                     args.fuzz_target, build_data)
+  if not error_sha:
+    print('No error was found in commit range %s:%s' %
+          (args.commit_old, args.commit_new))
+    return 1
+  print('Error was introduced at commit %s' % error_sha)
+  return 0
+
+
+def bisect(commit_old, commit_new, testcase, fuzz_target, build_data):
+  """From a commit range, this function caluclates which introduced a
+  specific error from a fuzz testcase.
+
+  Args:
+    commit_old: The oldest commit in the error regression range
+    commit_new: The newest commit in the error regression range
+    testcase: The file path of the test case that triggers the error
+    fuzz_target: The name of the fuzzer to be tested
+    build_data: a class holding all of the input parameters for bisection
+
+  Returns:
+    The commit SHA that introduced the error or None
+  """
+  local_store_path = tempfile.mkdtemp()
+  repo_url = build_specified_commit.infer_main_repo(build_data.project_name,
+                                                    local_store_path,
+                                                    commit_old)
+  bisect_repo_manager = repo_manager.RepoManager(repo_url, local_store_path)
+  commit_list = bisect_repo_manager.get_commit_list(commit_old, commit_new)
+  build_specified_commit.build_fuzzer_from_commit(
+      build_data.project_name, commit_list[0], bisect_repo_manager.repo_dir,
+      build_data.engine, build_data.sanitizer, build_data.architecture,
+      bisect_repo_manager)
+  error_code = helper.reproduce_impl(build_data.project_name, fuzz_target,
+                                     False, [], [], testcase)
+  old_idx = len(commit_list) - 1
+  new_idx = 0
+  if len(commit_list) == 1:
```

Comment:
This check is not needed. If the length of the list is 1, the while loop below won't be executed and return on the line 149 will return the correct revision

Reply:
Wont old_idx be 0 and new new_idx also be zero resulting in an infinite loop?